axelor-redmine/src/main/java/com/axelor/apps/redmine/sync/process/RedmineSyncProcessServiceImpl.java

```diff
@@ -0,0 +1,165 @@
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class RedmineSyncProcessServiceImpl implements RedmineSyncProcessService {
+
+  protected RedmineExportProjectService redmineExportProjectService;
+  protected RedmineExportVersionService redmineExportVersionService;
+  protected RedmineExportIssueService redmineExportIssueService;
+  protected RedmineExportTimeSpentService redmineExportTimeSpentService;
+  protected RedmineImportTrackerService redmineImportTrackerService;
+  protected RedmineImportProjectService redmineImportProjectService;
+  protected RedmineImportVersionService redmineImportVersionService;
+  protected RedmineImportIssueService redmineImportIssueService;
+  protected RedmineImportTimeSpentService redmineImportTimeSpentService;
+  protected RedmineFetchExportDataService redmineFetchExportDataService;
+  protected RedmineFetchImportDataService redmineFetchImportDataService;
+  protected BatchRepository batchRepo;
+
+  @Inject
+  public RedmineSyncProcessServiceImpl(
+      RedmineExportProjectService redmineExportProjectService,
+      RedmineExportVersionService redmineExportVersionService,
+      RedmineExportIssueService redmineExportIssueService,
+      RedmineExportTimeSpentService redmineExportTimeSpentService,
+      RedmineImportTrackerService redmineImportTrackerService,
+      RedmineImportProjectService redmineImportProjectService,
+      RedmineImportVersionService redmineImportVersionService,
+      RedmineImportIssueService redmineImportIssueService,
+      RedmineImportTimeSpentService redmineImportTimeSpentService,
+      RedmineFetchExportDataService redmineFetchExportDataService,
+      RedmineFetchImportDataService redmineFetchImportDataService,
+      BatchRepository batchRepo) {
+
+    this.redmineExportProjectService = redmineExportProjectService;
+    this.redmineExportVersionService = redmineExportVersionService;
+    this.redmineExportIssueService = redmineExportIssueService;
+    this.redmineExportTimeSpentService = redmineExportTimeSpentService;
+    this.redmineImportTrackerService = redmineImportTrackerService;
+    this.redmineImportProjectService = redmineImportProjectService;
+    this.redmineImportVersionService = redmineImportVersionService;
+    this.redmineImportIssueService = redmineImportIssueService;
+    this.redmineImportTimeSpentService = redmineImportTimeSpentService;
+    this.redmineFetchExportDataService = redmineFetchExportDataService;
+    this.redmineFetchImportDataService = redmineFetchImportDataService;
+    this.batchRepo = batchRepo;
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public void redmineSyncProcess(
+      Batch batch,
+      RedmineManager redmineManager,
+      Consumer<Object> onSuccess,
+      Consumer<Throwable> onError) {
+
+    RedmineExportService.result = "";
+    RedmineImportService.result = "";
+
+    // FETCH EXPORT AND IMPORT DATA
+
+    Batch lastBatch =
```

Comment:
Just lastBatch.updatedOn is required, so pass it only, whole batch is not required in any method.